src/sys/mman.rs

```diff
@@ -315,6 +315,28 @@ pub unsafe fn mmap(addr: *mut c_void, length: size_t, prot: ProtFlags, flags: Ma
     }
 }
 
+/// Expands (or shrinks) an existing memory mapping, potentially moving it at the same time.
+///
+/// # Safety
+///
+/// See the `mremap(2)` man page for detailed requirements.
+#[cfg(target_os = "linux")]
+pub unsafe fn mremap(
+    addr: *mut c_void,
+    old_size: size_t,
+    new_size: size_t,
+    flags: MapFlags,
+    new_address: size_t,
```

Comment:
Shouldn't `new_address` be optional?

Reply:
Yes; I wasn't sure the best way to handle this since Rust doesn't support variadic functions. The argument is ignored `MAP_FIXED` isn't in `flags`; i.e. if `MAP_FIXED` isn't set the caller can put anything there, and it'll be ignored.

Maybe it'd be a bit clearer to make it an `Option<size_t>`?

Reply:
I just realized the flags enum is wrong; will fix that and change new_address to be an `Option<size_t>` (but lmk if there's another solution you recommend)

Reply:
Done. Also added a few tests, though in hindsight maybe these should just be doc tests (and drop the second mremap test)?